[Vis Editor] Fix fields filtering in top hit agg

sulemanof opened the pull request:
## Summary

Is a part of https://github.com/elastic/kibana/issues/54460 . Fix is only acceptable for `master` & `7.x`.
Fix field types filtering in `Sort on` param inside the `Top Hit` aggregation.

![image](https://user-images.githubusercontent.com/31325372/73444492-b70e0d80-4369-11ea-85e6-3f047c7d01b9.png)


### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)
- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios
- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)
- [ ] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23477/)
* Commit: 81ce24a49d76706eb8e28291779ae7ba35780ea4

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23477/","number":"23477","commit":"81ce24a49d76706eb8e28291779ae7ba35780ea4"}]}
PIPELINE-->

sulemanof commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23932/)
* Commit: fc8995ba567a3cad749143c037cafcf0b14b51b5

### History
* :green_heart: [Build #23477](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23477/) succeeded 81ce24a49d76706eb8e28291779ae7ba35780ea4

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23477/","number":"23477","commit":"81ce24a49d76706eb8e28291779ae7ba35780ea4"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23932/","number":"23932","commit":"fc8995ba567a3cad749143c037cafcf0b14b51b5"}]}
PIPELINE-->

maryia-lapata commented:
@sulemanof 
I created a Table visualization and faced errors:

<img height="500" src="https://user-images.githubusercontent.com/11524637/73641083-7cb1b280-4680-11ea-9e6d-c3b6502de1c1.png" />

In some case it happens that [`field`](https://github.com/elastic/kibana/blob/master/src/legacy/ui/public/agg_types/metrics/top_hit.ts#L83) is undefined. Could you please investigate it?

elasticmachine commented:
Pinging @elastic/kibana-app (Team:KibanaApp)

sulemanof commented:
Hey @maryia-lapata 
Thanks for a good catch!
I've fixed it. The problem starts in `src/legacy/core_plugins/data/public/search/expressions/esaggs.ts`
while creating the `AggConfigs` without schemas... 
![image](https://user-images.githubusercontent.com/31325372/73662793-7a198200-46ad-11ea-9f7e-34aa6857b2d1.png)
In that case an `unknown` schema is used by default.
I've added default configs to the unknown schema, but I hope after refactoring of the `AggConfigs` this will gone.
@ppisljar could you please confirm these changes?

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24007/)
* Commit: f2a38032720c9d02000331016ff5b270419c9433

### History
* :green_heart: [Build #23932](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23932/) succeeded fc8995ba567a3cad749143c037cafcf0b14b51b5
* :green_heart: [Build #23477](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23477/) succeeded 81ce24a49d76706eb8e28291779ae7ba35780ea4

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23477/","number":"23477","commit":"81ce24a49d76706eb8e28291779ae7ba35780ea4"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23932/","number":"23932","commit":"fc8995ba567a3cad749143c037cafcf0b14b51b5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24007/","number":"24007","commit":"f2a38032720c9d02000331016ff5b270419c9433"}]}
PIPELINE-->

lukeelmers commented:
> I've added default configs to the unknown schema, but I hope after refactoring of the AggConfigs this will gone.

This makes sense to me, but I'd like @ppisljar to confirm too.

Also wanted to note there are a few places in this PR which will conflict with #56353, which I expect to merge any time now (waiting on one last LGTM).

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24569/)
* Commit: a1d0ae32519e2d7039f73ab79085be9738871884

### History
* :green_heart: [Build #24007](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24007/) succeeded f2a38032720c9d02000331016ff5b270419c9433
* :green_heart: [Build #23932](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23932/) succeeded fc8995ba567a3cad749143c037cafcf0b14b51b5
* :green_heart: [Build #23477](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23477/) succeeded 81ce24a49d76706eb8e28291779ae7ba35780ea4

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23477/","number":"23477","commit":"81ce24a49d76706eb8e28291779ae7ba35780ea4"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23932/","number":"23932","commit":"fc8995ba567a3cad749143c037cafcf0b14b51b5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24007/","number":"24007","commit":"f2a38032720c9d02000331016ff5b270419c9433"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24569/","number":"24569","commit":"a1d0ae32519e2d7039f73ab79085be9738871884"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24593/)
* Commit: 0476a0a72f5ce47478e4e299091752a82edd2eff

### History
* :green_heart: [Build #24569](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24569/) succeeded a1d0ae32519e2d7039f73ab79085be9738871884
* :green_heart: [Build #24007](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24007/) succeeded f2a38032720c9d02000331016ff5b270419c9433
* :green_heart: [Build #23932](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23932/) succeeded fc8995ba567a3cad749143c037cafcf0b14b51b5
* :green_heart: [Build #23477](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23477/) succeeded 81ce24a49d76706eb8e28291779ae7ba35780ea4

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23477/","number":"23477","commit":"81ce24a49d76706eb8e28291779ae7ba35780ea4"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23932/","number":"23932","commit":"fc8995ba567a3cad749143c037cafcf0b14b51b5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24007/","number":"24007","commit":"f2a38032720c9d02000331016ff5b270419c9433"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24569/","number":"24569","commit":"a1d0ae32519e2d7039f73ab79085be9738871884"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24593/","number":"24593","commit":"0476a0a72f5ce47478e4e299091752a82edd2eff"}]}
PIPELINE-->
